### PR TITLE
Add basic (mode 9 and 1000) mouse support

### LINKED ISSUE #15: Missing support for mouse events
I tested the [micro](https://github.com/zyedidia/micro) editor, which I use quite frequently, and it does not work in fyneterm. Visually, it seems to looks mostly as expected with just the colour behind the line numbers missing (see comparison images below). It is however, not possible to click anywhere inside it or type anything in to it.

This is the printout to the other terminal as it is being loaded:
```
2021/05/28 19:18:48 Unknown private escape code ?2004[hl]
2021/05/28 19:19:38 Unknown private escape code ?2004[hl]
2021/05/28 19:19:38 Unknown private escape code ?2004[hl]
2021/05/28 19:20:02 Unknown private escape code ?2004[hl]
2021/05/28 19:20:02 Unknown private escape code ?2004[hl]
2021/05/28 19:20:11 Unknown private escape code ?2004[hl]
2021/05/28 19:20:11 Unknown private escape code ?2004[hl]
2021/05/28 19:20:19 Unknown private escape code ?2004[hl]
2021/05/28 19:20:19 Unknown private escape code ?2004[hl]
2021/05/28 19:20:46 Unknown private escape code ?2004[hl]
2021/05/28 19:20:46 Unrecognised Escape: 22;0;0t
2021/05/28 19:20:46 Unknown private escape code ?2004[hl]
2021/05/28 19:20:46 Unknown private escape code ?1000[hl]
2021/05/28 19:20:46 Unknown private escape code ?1002[hl]
2021/05/28 19:20:46 Unknown private escape code ?1006[hl]
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 236
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unsupported graphics mode 48
2021/05/28 19:20:46 Unsupported graphics mode 5
2021/05/28 19:20:46 Unsupported graphics mode 235
2021/05/28 19:20:46 Unknown private escape code ?12[hl]
```

## Micro in fyneterm:
image

## Micro in gnome terminal:
image

### PR DESCRIPTION
**Pull request by andydotxyz**
Adding simple support for basic mouse event reporting

Fixes #15

### CODE REVIEW

File: mouse.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package terminal
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/driver/desktop"
+)
+
+func (t *Terminal) handleMouseDownV200(btn int, mods desktop.Modifier, pos fyne.Position) {
+	_, _ = t.Write(t.encodeMouse(btn, mods, pos))
```
Comment (Jacalz) [inline mouse.go:9]: Hmm, I am curious as to why staticcheck doesn't complain about this. I think it's fine to just do `t.Write()` without the blank identifier assignments. This applies to the below items as well.

**Review by Jacalz — CHANGES_REQUESTED** [review]
This is working very well inside micro. Thank you :)
Just a little comment on code style in one place. I also have a little though.

This does not seem to make it possible to change the cursor position with the mouse as you are just typing commands into the shell and does not add selection with the mouse. Should that be part of different PR (given that selection isn't doable with keyboard at the moment)?

File: mouse.go (hunk 1 above)
Comment (andydotxyz) [reply mouse.go:9]: If I leave out the assignment then the IDE warns about ignored errors

**Comment by andydotxyz** [comment]
> Should that be part of different PR (given that selection isn't doable with keyboard at the moment)?

Yes, a separate PR will be required after we write selection support in the main app.

**Review by Jacalz — APPROVED** [review]

**Pull request merged**